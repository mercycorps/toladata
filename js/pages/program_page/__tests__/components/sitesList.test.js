import SitesList from '../../components/sitesList';
import React from 'react';
import TestRenderer from 'react-test-renderer';

describe('Sites List', () => {
    it("shows add site button", () => {
        const rootStore = {program: {siteCount: 1}};
        const tree = TestRenderer.create(<SitesList rootStore={rootStore} />);
        expect(tree.toJSON()).toMatchSnapshot();
        expect(tree.root.findByProps({className: 'btn-link text-success'}).children).toContain("Add site");
    });
    it("shows no sites message when there are no sites", () => {
        const rootStore = {program: {siteCount: 0}};
        const tree = TestRenderer.create(<SitesList rootStore={rootStore} />);
        expect(tree.toJSON()).toMatchSnapshot();
        expect(tree.root.findByProps({className: 'text-muted'}).children).toContain("There are no program sites.");
    });
    it("shows view sites link when there are sites", () => {
        const rootStore = {program: {siteCount: 1}};
        const tree = TestRenderer.create(<SitesList rootStore={rootStore} />);
        expect(tree.toJSON()).toMatchSnapshot();
        expect(tree.root.findByType('ul').children[0].children[0].children[0]).toContain("View program sites");
    });
})
