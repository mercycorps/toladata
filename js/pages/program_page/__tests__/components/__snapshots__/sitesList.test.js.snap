// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sites List shows add site button 1`] = `
Array [
  <h3>
    Sites
  </h3>,
  <ul
    className="list-unstyled"
  >
    <li>
      <a
        href="/workflow/siteprofile_list/undefined/0/"
      >
        View program sites
      </a>
    </li>
    <li
      className="pt-2"
    >
      <a
        className="btn-link text-success"
        href="/workflow/siteprofile_add/"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus-circle fa-w-16 "
          data-icon="plus-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        Add site
      </a>
    </li>
  </ul>,
]
`;

exports[`Sites List shows no sites message when there are no sites 1`] = `
Array [
  <h3>
    Sites
  </h3>,
  <ul
    className="list-unstyled"
  >
    <li
      className="text-muted"
    >
      There are no program sites.
    </li>
    <li
      className="pt-2"
    >
      <a
        className="btn-link text-success"
        href="/workflow/siteprofile_add/"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus-circle fa-w-16 "
          data-icon="plus-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        Add site
      </a>
    </li>
  </ul>,
]
`;

exports[`Sites List shows view sites link when there are sites 1`] = `
Array [
  <h3>
    Sites
  </h3>,
  <ul
    className="list-unstyled"
  >
    <li>
      <a
        href="/workflow/siteprofile_list/undefined/0/"
      >
        View program sites
      </a>
    </li>
    <li
      className="pt-2"
    >
      <a
        className="btn-link text-success"
        href="/workflow/siteprofile_add/"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus-circle fa-w-16 "
          data-icon="plus-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        Add site
      </a>
    </li>
  </ul>,
]
`;
